executor/point_get_test.go

```diff
@@ -594,3 +595,52 @@ func (s *testPointGetSuite) TestClusterIndexCBOPointGet(c *C) {
 		res.Check(testkit.Rows(output[i].Res...))
 	}
 }
+
+func (s *testPointGetSuite) TestPointGetReadLock(c *C) {
+	config.UpdateGlobal(func(conf *config.Config) {
+		conf.EnableTableLock = true
+	})
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("create table point (id int primary key, c int, d varchar(10), unique c_d (c, d))")
+	tk.MustExec("insert point values (1, 1, 'a')")
+	tk.MustExec("insert point values (2, 2, 'b')")
+	tk.MustExec("lock tables point read")
+	result, err := tk.Exec("select id as ident from point where id = 1")
+	c.Assert(err, IsNil)
+	fields := result.Fields()
+	c.Assert(fields[0].ColumnAsName.O, Equals, "ident")
+	result.Close()
+	tk.MustQuery(`select * from point where id = 1;`).Check(testkit.Rows(
```

Comment:
How to make sure the cache is taking effect?
recommend to check the explain analyze result:
```sql
mysql>create table point (id int primary key, c int, d varchar(10), unique c_d (c, d))
      ;
Query OK, 0 rows affected
Time: 0.005s
mysql>insert point values (1, 1, 'a');
Query OK, 1 row affected
Time: 0.003s
mysql>insert point values (2, 2, 'b');
Query OK, 1 row affected
Time: 0.001s
mysql>explain analyze select * from t where a=1;
+-------------+---------+---------+------+---------------+-------------------------------------------------------------+---------------+--------+------+
| id          | estRows | actRows | task | access object | execution info                                              | operator info | memory | disk |
+-------------+---------+---------+------+---------------+-------------------------------------------------------------+---------------+--------+------+
| Point_Get_1 | 1.00    | 1       | root | table:t       | time:24.046µs, loops:2, Get:{num_rpc:1, total_time:8.456µs} | handle:1      | N/A    | N/A  |
+-------------+---------+---------+------+---------------+-------------------------------------------------------------+---------------+--------+------+
1 row in set
Time: 0.007s
mysql>lock table t read;
Query OK, 0 rows affected
Time: 0.007s
mysql>explain analyze select * from t where a=1;
+-------------+---------+---------+------+---------------+-------------------------------------------------------------+---------------+--------+------+
| id          | estRows | actRows | task | access object | execution info                                              | operator info | memory | disk |
+-------------+---------+---------+------+---------------+-------------------------------------------------------------+---------------+--------+------+
| Point_Get_1 | 1.00    | 1       | root | table:t       | time:28.861µs, loops:2, Get:{num_rpc:1, total_time:8.159µs} | handle:1      | N/A    | N/A  |
+-------------+---------+---------+------+---------------+-------------------------------------------------------------+---------------+--------+------+
1 row in set
Time: 0.007s
mysql>explain analyze select * from t where a=1;
+-------------+---------+---------+------+---------------+-------------------------+---------------+--------+------+
| id          | estRows | actRows | task | access object | execution info          | operator info | memory | disk |
+-------------+---------+---------+------+---------------+-------------------------+---------------+--------+------+
| Point_Get_1 | 1.00    | 1       | root | table:t       | time:5.103µs, loops:2,  | handle:1      | N/A    | N/A  |
+-------------+---------+---------+------+---------------+-------------------------+---------------+--------+------+
1 row in set
```

check the `execution info`, as you can see, after `lock tables point read`, the first execution of `select * from point where id = 1;` still has rpc information, it means that was read from TiKV, but the second time, the rpc information was gone, it means it doesn't send rpc to tikv, also means it read from the cache.

Reply:
thanks.

Reply:
address